typhos/positioner.py

```diff
@@ -654,18 +730,266 @@ def update_alarm_text(self, alarm_level):
+        AlarmLevel.DISCONNECTED: 'conn',
+        AlarmLevel.INVALID: 'inv',
+    }
+
+    def __init__(self, *args, **kwargs):
+        self._error_message = ""
+        self._status_text = ""
+        self._alarm_level = AlarmLevel.DISCONNECTED
+
+        super().__init__(*args, **kwargs)
+
+        for idx in range(self.layout().count()):
+            item = self.layout().itemAt(idx)
+            if item is self.ui.status_text_layout:
+                self.layout().takeAt(idx)
+                break
+
+        # TODO move these out
+        self._omit_names = [
+            "motor_egu",
+            "motor_stop",
+            "motor_done_move",
+            "direction_of_travel",
+            "user_readback",
+            "user_setpoint",
+            "home_forward",  # maybe keep?
+            "home_reverse",
+        ]
+
+        self.ui.extended_signal_panel = None
+        self.ui.expand_button.clicked.connect(self._expand_layout)
+        self.ui.status_label.setText("")
+
+        # TODO: ${name} / macros don't expand here
+
+    @QtCore.Property("QStringList")
+    def omitNames(self) -> list[str]:
+        """Get or set the list of names to omit in the expanded signal panel."""
+        return self._omit_names
+
+    @omitNames.setter
+    def omitNames(self, omit_names: list[str]) -> None:
+        if omit_names == self._omit_names:
+            return
+
+        self._omit_names = list(omit_names or [])
+        if self.ui.extended_signal_panel is not None:
+            self.ui.extended_signal_panel.omitNames = self._omit_names
+
+    def get_names_to_omit(self) -> list[str]:
+        """
+        Get a list of signal names to omit in the extended panel.
+
+        Returns
+        -------
+        list[str]
+        """
+        device: Optional[ophyd.Device] = self.device
+        if device is None:
+            return []
+
+        omit_signals = self.all_linked_signals
+        to_keep_signals = [
+            getattr(device, attr, None)
+            for attr in (self.velocity_attribute, self.acceleration_attribute)
+        ]
+        for sig in to_keep_signals:
+            if sig in omit_signals:
+                omit_signals.remove(sig)
+
+        to_omit = set(sig.name for sig in omit_signals)
+
+        # TODO: move these to a Qt designable property
+        for name in self.omitNames:
+            to_omit.add(name)
+
+        if device.name in to_omit:
+            # Don't let the renamed position signal stop us from showing any
+            # signals:
+            to_omit.remove(device.name)
+        return sorted(to_omit)
+
+    def _create_signal_panel(self) -> Optional[TyphosSignalPanel]:
+        """Create the 'extended' TyphosSignalPanel for the device."""
+        if self.device is None:
+            return None
+
+        panel = TyphosSignalPanel()
+        panel.omitNames = self.get_names_to_omit()
+        panel.sortBy = SignalOrder.byName
+        panel.add_device(self.device)
```

Comment:
missing is here is filtering the new signal panel by the kinds specified in the menu above

interactively right now, if you deselect e.g. config or add omitted it only applies to existing signal panels

Reply:
Good catch - that needs addressing

Reply:
In more words: the filters work, but they only apply to positioner rows that have already been expanded at least once, because otherwise the signal panel has never been created

Reply:
I'll copy my original comment now that I fully understand - "good catch - that needs addressing!" 😁 

I think the linked code will have to be modified to tell the positioner widget (when it's ready) how to create the signal panel with the appropriate kind settings

Reply:
fwiw I think this is a small thing and could be ignored and doubled-back on later
the main drawback of the status quo is that all the omitted signals appear for the first expansion

Reply:
#579